### PR TITLE
Add religion data

### LINKED ISSUE #169: Implement town religiosity
Town religiosity slider, similar to the currently implemented town welfare. Should ideally impact the likelihood of religion sector professions, and have knock on effects such as changing the town welfare roll; most charity was pushed by religion in medieval times.

### PR DESCRIPTION
**Pull request by ryceg**
## What does this do?

- Adds a religion description based on the NPC's religiosity and gregariousness.
- Adds town religion (they worship specific saints, and only non-conforming NPCs will worship different ones)

## How was this tested? Did you test the changes in the compiled `.html` file?

`.html`

## Is there a [GitHub Issue](https://github.com/ryceg/Eigengrau-s-Essential-Establishment-Generator/issues?q=is%3Aopen+is%3Aissue) that this is resolving?

Closes #169.

### CODE REVIEW

**Comment by Glinkis** [comment]
Wait, did I bork this?

**Comment by ryceg** [comment]
I think that there was a version of the file that perhaps got overwritten at some point, but this is exactly the same, so there's no harm. Still want to do a little bit more stuff, so that's why it's not merged.

**Comment by Glinkis** [comment]
Import from `@lib`, instead of individual files. Also prefer `import type {}` when possible.

File: lib/religion/religion.ts [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+    cowardly: 'valorous'
+  }
+}
+
+export type Virtues = keyof typeof traits['virtueKey']
+export type Vices = keyof typeof traits['viceKey']
+export type VirtuesVices = Virtues | Vices
+
+export interface Pantheon {
+  /** The name of the patheon, without riders or indefinite articles.
```
Comment (Glinkis) [inline lib/religion/religion.ts:54]: Linne break here when these comments are multi-line.

File: lib/religion/religion.ts [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+     * @example Zeus: 'lightning'
+     * @usage 'In combat, Zeus uses `lightning`.
+     */
+    weapon: string
+    tactics: string
+  }
+  /** For things that the deity owns.
+   * @example `${'Thor'} owns the ${'hammer'} ${'Mjölnir'}, which ${"could return to its owner's hand when thrown, and call lightning down on enemies."}`
+   */
+  possessions: {
```
Comment (Glinkis) [inline lib/religion/religion.ts:125]: Perhaps extract a `Possession` in interface.

File: lib/religion/religion.ts [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+   *   lust: 80
+   * }```
+   */
+  // FIXME can't
+  // personality: Record<PartialVirtues, number>
+  personality: Record<string, number>
+  /** Manifesting an avatar can be energy intensive; deities that wish to announce their presence may sometimes do so through a more economic way, by means of a supernatural event, such as an item glowing a specific colour, a duck appearing out of nowhere, etc. */
+  manifestations: {
+    /** A deity can have multiple different avatars, some more rare than others. */
+    avatars: {
```
Comment (Glinkis) [inline lib/religion/religion.ts:145]: Perhaps add an `Avatar` interface?

Comment (ryceg) [reply lib/religion/religion.ts:145]: Good idea!

**Review by BGrayy — APPROVED** [review]

**Pull request merged**